Doc/library/typing.rst

```diff
@@ -933,6 +933,55 @@ These can be used as types in annotations using ``[]``, each having a unique syn
 
    .. versionadded:: 3.9
 
+
+.. data:: TypeGuard
+
+   Special typing form used to annotate the return type of a user-defined
+   type guard function.  ``TypeGuard`` only accepts a single type argument.
```

Comment:
I'd add right here something like this:
```suggestion
   type guard function.  ``TypeGuard`` only accepts a single type argument.
   At runtime, functions marked this way return a boolean.
```
This would hopefully allay the repeatedly expressed worries that writing `-> TypeGuard[X]` obfuscates that the return type of such functions is `bool`.

(Maybe ``` (i.e., ``True`` or ``False``)``` is redundant.)